### PR TITLE
fix : filter enabled while searching

### LINKED ISSUE #978: Filters still enabled after Search in Accounts Section
Summarize your issue in one sentence (what goes wrong, what did you expect to happen).
In Accounts, the filters are still clicked, even after the user uses search option, which basically removes all the filters.
**Steps to reproduce:**
1. Apply any filter.(Suppose closed)
2. Use search and type something in search bar, then cancel the search option (List is now unfiltered)
3. Open filters
You will see that "closed" is still checked.
**Expected behaviour:**
Filters should be removed as the list is now showing unfiltered items.

### PR DESCRIPTION
**Pull request by shiv07tiwari**
Fixes #978

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by shiv07tiwari** [comment]
@miPlodder @luckyman20 @therajanmaurya Kindly review this PR.

**Comment by luckyman20** [comment]
Kindly add a GIF showing your changes.

File: app/src/main/java/org/mifos/mobilebanking/ui/fragments/AccountsFragment.java [hunk 1]
```diff
@@ -299,29 +299,77 @@ public void showEmptyAccounts(String emptyAccounts) {
...
     /**
      * Used for searching an {@code input} String in {@code loanAccounts} and displaying it in the
      * recyclerview.
      * @param input String which is needs to be searched in list
      */
-    public void searchLoanAccount(String input) {
+    public void searchLoanAccount(String input, List<CheckboxStatus> statusModelList) {
+        List<LoanAccount> loanAccountSearchList = loanAccounts;
+        if (statusModelList != null) {
+            for (int i = 0; i < statusModelList.size(); i++) {
```
Comment (luckyman20) [inline app/src/main/java/org/mifos/mobilebanking/ui/fragments/AccountsFragment.java]: Use `foreach` instead. It will improve efficiency of the program.

**Comment by shiv07tiwari** [comment]
Made the changes @luckyman20

File: app/src/main/java/org/mifos/mobilebanking/ui/fragments/AccountsFragment.java [hunk 2]
```diff
@@ -299,29 +299,77 @@ public void showEmptyAccounts(String emptyAccounts) {
...
+        List<SavingAccount> savingAccountSearchList = savingAccounts;
+        if (statusModelList != null) {
+            for (CheckboxStatus statusList: statusModelList) {
+                if (statusList.component3()) {
+                    List<SavingAccount> filteredSavings = new ArrayList<>();
+                    for (CheckboxStatus status :
+                            accountsPresenter.getCheckedStatus(statusModelList)) {
+                        filteredSavings.addAll(
+                                accountsPresenter.getFilteredSavingsAccount(savingAccounts,
+                                        status));
```
Comment (luckyman20) [inline app/src/main/java/org/mifos/mobilebanking/ui/fragments/AccountsFragment.java]: We can minimize the code lines here. We can remove all the elements of `savingAccountSearchList` and then add the elements here.

**Comment by shiv07tiwari** [comment]
@luckyman20  Made the changes.

**Review by luckyman20 — APPROVED** [review]
Looks good to me. @miPlodder Please do a final review and then merge it.

**Comment by shiv07tiwari** [comment]
@miPlodder @therajanmaurya Kindly review this PR.

**Comment by shiv07tiwari** [comment]
@therajanmaurya @miPlodder Kindly review this.

**Comment by shiv07tiwari** [comment]
@therajanmaurya @miPlodder Kindly review the PR

**Comment by shiv07tiwari** [comment]
@therajanmaurya Fixed the merge conflict.

**Comment by prashantkh19** [comment]
@shiv07tiwari did you try running tests on your local machine?

**Comment by prashantkh19** [comment]
@shiv07tiwari rebase the branch and fix the travis build error.

**Comment by jawidMuhammadi** [comment]
This PR has conflicts and the issue already has been resolved by another PR, therefore, closing it.

**Pull request closed without merging**